src/python/dxpy/sugar/objects.py

```diff
@@ -0,0 +1,454 @@
+            workflow.describe()
+            return workflow
+        except dxpy.DXError:
+            pass
+
+    kwargs["limit"] = 2
+    candidates = dxpy.find_global_workflows(**kwargs)
+    workflow = next(candidates, None)
+    if workflow is None:
+        raise dxpy.DXSearchError(f"Expected one result, but found none: {kwargs}")
+    if next(candidates, None) is not None:
+        raise dxpy.DXSearchError(f"Expected one result, but found more: {kwargs}")
+
+    return workflow
+
+
+def get_app_or_applet(
+    app_or_applet_desc: Union[str, dxpy.DXApp, dxpy.DXApplet],
+    project: Optional[str] = None,
+) -> Union[dxpy.DXApp, dxpy.DXApplet]:
+    """
+    Gets an app(let) by its ID or name. This is a convenience method for resolving an a app(let)
+    when, for example, a user can provide either an applet or an app ID/name as a paramter to an
+    app.
+
+    Args:
+        app_or_applet_desc: ID or name of the app/applet, or an instance of DXApp or DXApplet.
+        project: name, ID, or DXProject object of project containing applet.
+
+    Returns:
+        DXApp or DXApplet object.
+
+    Raises:
+        * dxpy.DXSearchError if the app(let) does not exist.
+        * dxpy.exceptions.PermissionDenied if user does not have proper permissions
+            to access the app(let).
+    """
+    if isinstance(app_or_applet_desc, (dxpy.DXApp, dxpy.DXApplet)):
+        return app_or_applet_desc
+
+    if app_or_applet_desc.startswith("applet-"):
+        app = get_data_object(app_or_applet_desc, project, classname="applet")
+        if app is None:
+            app = get_app(app_or_applet_desc)
+    else:
+        app = get_app(app_or_applet_desc)
+        if app is None:
+            app = get_data_object(app_or_applet_desc, project, classname="applet")
+
+    if app is None:
+        raise dxpy.DXSearchError(
+            f"Did not find exactly one app(let): {app_or_applet_desc}"
+        )
+
+    return app
+
+
+def get_app(
+    app_desc: Union[str, dxpy.DXApp], version: Optional[str] = None, **kwargs
+) -> dxpy.DXApp:
+    """
+    Gets the app with the given ID or name.
+
+    Args:
+        app_desc: The app ID or name, or a DXApp instance.
+        version: App version.
+        kwargs: Additional kwargs to use when looking up the app by name.
+
+    Returns:
+        A DXApp object.
+
+    Raises:
+        * dxpy.DXSearchError if the app does not exist.
+        * dxpy.exceptions.PermissionDenied if user does not have permissions to access the app.
+    """
+    if isinstance(app_desc, dxpy.DXApp):
+        return app_desc
+
+    if resolver.is_hashid(app_desc):
+        try:
+            LOG.info("Checking if app with ID %s exists", app_desc)
+            executable = dxpy.DXApp(app_desc)
+            executable.describe()
+            return executable
+        except dxpy.DXError:
+            pass
+
+    if app_desc.startswith("app-"):
+        app_desc = app_desc[4:]
+
+    if "/" in app_desc:
```

Comment:
The same should be applied to globalworkflows. It is also possible to refer to it using name/version.

Reply:
thanks, this is done